### PR TITLE
Fixed an issue where `firestore:delete `would not use the emulator when it should

### LINKED ISSUE #6537: firestore:delete not working against emulator since version 12.6.2 / fails in 12.7+
### [REQUIRED] Environment info

**firebase-tools:** 12.9.1

**Platform:** any platform, mostly testing macOS

### [REQUIRED] Test case

Note that **this will work against firestore in the cloud if not careful** please do not destroy data accidentally!

It does not delete data against the emulator but against the cloud it works just fine! 

- Start the firestore emulator
- Create a firestore collection in your emulator
- run `FIRESTORE_EMULATOR_HOST=localhost:8080 yarn firebase firestore:delete --all-collections --project <your project name> --account <your acct> --force`

(or, at least I run it with yarn around it, but I'm sure you see how to run it - project name / account are whatever you need in your local env against your emulator)

### [REQUIRED] Steps to reproduce

The test case above is it, if you call this using firebase-tools 12.6.2, all the collections will be removed. From 12.7.0 onwards the collections will not be removed.

The emulator server side seems version-insensitive, the behavior varies by version on the calling side

### [REQUIRED] Expected behavior

All the collections should be removed, for a fresh data load (I test data loads + migrations...)

### [REQUIRED] Actual behavior

The collections stay in place, no data is deleted.

I bisected the versions to see which version began failing because I knew it used to work, but it took me a while to find time to pinpoint this for this repo (I'm normally swamped, as the react-native-firebase maintainer 👋 😆 )

```

firebase-tools 10.2.2 calling to 12.9.1 works (use node v16 to work)
firebase-tools ^10 (10.9.2) calling to 12.9.1 works (use node v16 to work)
firebase-tools ^11 (11.30.0) callnig to 12.9.1 works (use node v16 to work)
Firebase-tools ^12 (12.9.1) calling to 12.9.1 does not work (v16, v18 or v20 node all fail)
callnig with 12.0.0 to 12.9.1 works
~12.5 (12.5.4) works
~12.7 (12.7.0) does not work
12.6.2 works
```

diff between 12.6.2 and 12.7.0 is the problem --> https://github.com/firebase/firebase-tools/compare/v12.6.2...v12.7.0

### PR DESCRIPTION
**Pull request by joehan**
### Description
Fixed an issue where `firestore:delete `would not use the emulator when `FIRESTORE_EMULATOR_HOST` was set
Fixes #6537

### CODE REVIEW

File: src/gcp/firestore.ts [hunk 1]
```diff
@@ -31,7 +37,12 @@ export interface Database {
  * @param {string} project the Google Cloud project
  * @param {string} database the Firestore database name
  */
-export async function getDatabase(project: string, database: string): Promise<Database> {
+export async function getDatabase(
+  project: string,
+  database: string,
+  allowEmualtor: boolean = false,
```
Comment (rwhogg) [inline src/gcp/firestore.ts:43]: Typo in the parameter name

File: src/gcp/firestore.ts [hunk 2]
```diff
@@ -71,7 +86,8 @@ export function listCollectionIds(project: string): Promise<string[]> {
  * @param {object} doc a Document object to delete.
  * @return {Promise} a promise for the delete operation.
  */
-export async function deleteDocument(doc: any): Promise<any> {
+export async function deleteDocument(doc: any, allowEmualtor: boolean = false): Promise<any> {
```
Comment (rwhogg) [inline src/gcp/firestore.ts:89]: Same typo here

File: src/gcp/firestore.ts [hunk 3]
```diff
@@ -85,7 +101,12 @@ export async function deleteDocument(doc: any): Promise<any> {
  * @param {object[]} docs an array of Document objects to delete.
  * @return {Promise<number>} a promise for the number of deleted documents.
  */
-export async function deleteDocuments(project: string, docs: any[]): Promise<number> {
+export async function deleteDocuments(
+  project: string,
+  docs: any[],
+  allowEmualtor: boolean = false,
```
Comment (rwhogg) [inline src/gcp/firestore.ts:107]: Ditto

File: src/gcp/firestore.ts [hunk 4]
```diff
@@ -2,7 +2,13 @@ import { firestoreOrigin } from "../api";
...
+const prodOnlyClient = new Client({
+  auth: true,
+  apiVersion: "v1",
+  urlPrefix: firestoreOrigin,
+});
+
+const emuOrProdClient = new Client({
   auth: true,
   apiVersion: "v1",
   urlPrefix: firestoreOrigin,
```
Comment (rwhogg) [inline src/gcp/firestore.ts:14]: Was this supposed to be `firestoreOriginOrEmulator`?

Comment (joehan) [reply src/gcp/firestore.ts:14]: 🤦 Thanks for catching this, I clearly rushed this one

**Review by rwhogg — APPROVED** [review]

**Comment by Pickachu** [comment]
Thanks for PR team!

**Pull request merged**